automation/roles/patroni/tasks/main.yml

```diff
@@ -427,6 +427,22 @@
         state: directory
         mode: "0700"
 
+    # Generating TLS Certificates
+    - name: Generating TLS Certificates to {{ postgresql_home_dir }}/{{ postgresql_version }}
+      become: true
+      become_user: postgres
+      ansible.builtin.command:
+        cmd: openssl req -nodes -new -x509 -days 3650 -keyout server.key -out server.crt -subj '/C=AL/L=City/O=Org/CN=PostgreSQL'
+        chdir: "{{ postgresql_home_dir }}/{{ postgresql_version }}/"
```

Comment:
Using the commands for generating certificates using openssl is really not an idempotent approach. Ansible has special modules that allow you to generate certificates

- https://docs.ansible.com/ansible/latest/collections/community/crypto/openssl_privatekey_module.html
- https://docs.ansible.com/ansible/latest/collections/community/crypto/x509_certificate_module.html

Something like this:

```yaml
- name: Generate a private key for TLS
  community.crypto.openssl_privatekey:
    path: "{{ postgresql_home_dir }}/server.key"
    size: 2048
    owner: postgres
    group: postgres
    mode: '0600'

- name: Generate a self-signed certificate for Postgres
  community.crypto.x509_certificate:
    path: "{{ postgresql_home_dir }}/server.crt"
    privatekey_path: "{{ postgresql_home_dir }}/server.key"
    owner: postgres
    group: postgres
    mode: '0644'
    provider: selfsigned
    subject:
      C: "AL"
      L: "City"
      O: "Org"
      CN: "PostgreSQL"
    valid_days: 3650
```

Reply:
This is a better approach. Will update it.